### PR TITLE
A17 and A27ab added

### PR DESCRIPTION
**Pull request by ianlongden**

### CODE REVIEW

File: src/chado_object/aberration/break_point.py [hunk 1]
```diff
@@ -48,6 +48,9 @@ def add_props(self, key, feature, position):
     for new_key in ['A90h', 'A90j', 'A90b']:
         if new_key == 'A90b':
             value = "{}_r{}:{}..{}".format(position['arm'].name, position['release'], position['start']-1, position['end'])
+            if position['strand']:
+                log.debug("strand is {}".format(position['strand']))
+                value += "-{}".format(position['strand'])
```
Comment (gildossantos) [inline src/chado_object/aberration/break_point.py:53]: I think this is for new "reported_genomic_loc" featureprops. In this case, when strand is indicated (n=4), it's appended as "(-)" or "(+)".
It's not clear to me that `position[strand]` value will be either a `-` or `+`. Can you point me to the code?
Also, that `+/-` char should be between brackets, not attached after a `-` char, as you have here.

File: src/chado_object/aberration/chado_aberration.py [hunk 2]
```diff
@@ -76,6 +78,34 @@ def __init__(self, params):
         self.checks_for_key = {'A26':  self.so_alter_check,
                                'A9': self.so_alter_check}
 
+    def phen_desc(self, key):
+        # get/create the genotype with same name as aberation
+        env, _ = get_or_create(self.session, Environment, uniquename=self.process_data[key]['environment'])
+        geno, _ = get_or_create(self.session, Genotype, uniquename=self.feature.name)
```
Comment (gildossantos) [inline src/chado_object/aberration/chado_aberration.py:84]: Need to use sgml for genotype uniquenames, not feature names. Brackets have to be converted to `<up>/</up>`, greek chars need to be in `&agr;`-type form, etc.
e.g., for genotype_id=372049 for aberration `Df(1)Vinc[2]`, we have genotype.uniquename = `Df(1)Vinc<up>2</up>`.

File: src/chado_object/aberration/chado_aberration.py [hunk 3]
```diff
@@ -76,6 +78,34 @@ def __init__(self, params):
...
+        desc_cvterm = get_cvterm(self.session, self.process_data[key]['desc_cv'], self.process_data[key]['desc_cvterm'])
+        fg_cvterm = get_cvterm(self.session, self.process_data[key]['cv'], self.process_data[key]['cvterm'])
+        chrom_cvterm = get_cvterm(self.session, self.process_data[key]['chrom_cv'], self.process_data[key]['chrom_cvterm'])
+        chrom, _ = get_or_create(self.session, Feature,
```
Comment (gildossantos) [inline src/chado_object/aberration/chado_aberration.py:88]: I don't think you should be creating chrom, here or anywhere, ever. We only make new chrom with new reference assembly scaffolds.

File: src/chado_object/aberration/chado_aberration.py [hunk 4]
```diff
@@ -76,6 +78,34 @@ def __init__(self, params):
...
+                                 name=self.process_data[key]['chrom'],
+                                 uniquename=self.process_data[key]['chrom'],
+                                 type_id=chrom_cvterm.cvterm_id)
+
+        # Add feature genotype
+        fg, _ = get_or_create(self.session, FeatureGenotype,
+                              chromosome_id=chrom.feature_id,
```
Comment (gildossantos) [inline src/chado_object/aberration/chado_aberration.py:95]: For all feature_genotypes, we have only one chromosome_id: `feature_id=23159230, name=unspecified`.

**Review by gildossantos — CHANGES_REQUESTED** [review]

File: src/chado_object/aberration/chado_aberration.py (hunk 3 above)
Comment (gildossantos) [reply src/chado_object/aberration/chado_aberration.py:88]: My mistake. I was confusing new "chromosome" features with "golden_path". The latter are associated with reference genome assemblies and should not be made via proforma. But the former can be created by proforma (though rare). So this is ok.

File: src/chado_object/aberration/chado_aberration.py (hunk 4 above)
Comment (gildossantos) [reply src/chado_object/aberration/chado_aberration.py:95]: Just talked to Ian. This is ok, as in perl parser and in the yaml files (new python), the chrom is hard-coded as "unspecified".

File: src/chado_object/aberration/break_point.py (hunk 1 above)
Comment (ianlongden) [reply src/chado_object/aberration/break_point.py]: Warning is now given and the strand is ignored.

**Review by gildossantos — APPROVED** [review]

**Pull request merged**